c7n/actions.py

```diff
@@ -460,16 +460,20 @@ def send_data_message(self, message):
 
     def send_sns(self, message):
         topic = self.data['transport']['topic']
-        region = topic.split(':', 5)[3]
+        region = self.manager.config.region
+        owner_id = self.manager.config.account_id
+        topic_arn="arn:aws:sns:"+region+":"+owner_id+":"+topic
```

Comment:
to preserve compatibility this should only be done if the topic  and queue values should be modified only if is their not already an arn value.